File: JS/Tema1_Ejercicio7.js
/*
 * TEMA 1: INTRODUCCIÓN BÁSICA - EJERCICIO 7
 * ===========================================
 * Calcular IMC (Índice de Masa Corporal) y clasificarlo.
 * 
 * EXPLICACIÓN:
 * - Fórmula: IMC = peso / (altura²)
 * - Condicionales anidadas para clasificar IMC
 * - Rangos de IMC: Bajo peso, Normal, Sobrepeso, Obeso
 */

function calcularIMC(peso, altura) {
    return peso / (altura * altura);
}

function clasificarIMC(imc) {
    if (imc < 18.5) return "Bajo peso";
    if (imc < 25) return "Normal";
    if (imc < 30) return "Sobrepeso";
    return "Obeso";
}

const peso = 70;  // kg
const altura = 1.75;  // metros

const imc = calcularIMC(peso, altura);
const clasificacion = clasificarIMC(imc);

console.log("=== CÁLCULO DE IMC ===");
console.log(`Peso: ${peso} kg`);
console.log(`Altura: ${altura} m`);
console.log(`IMC: ${imc.toFixed(2)}`);
console.log(`Clasificación: ${clasificacion}`);

/*
 * SALIDA ESPERADA:
 * === CÁLCULO DE IMC ===
 * Peso: 70 kg
 * Altura: 1.75 m
 * IMC: 22.86
 * Clasificación: Normal
 */

File: Java/Tema1_Ejercicio7.java
/*
 * TEMA 1: INTRODUCCIÓN BÁSICA - EJERCICIO 7
 * ==========================================
 * Escriba un programa que calcule el índice de masa corporal (IMC) 
 * de una persona dado su peso en kg y altura en metros.
 * 
 * EXPLICACIÓN:
 * - IMC = peso / (altura²)
 */

import java.util.Scanner;

public class Tema1_Ejercicio7 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        System.out.print("Digite su peso en kg: ");
        double peso = scanner.nextDouble();
        
        System.out.print("Digite su altura en metros: ");
        double altura = scanner.nextDouble();
        
        // Calcular IMC
        double imc = peso / (altura * altura);
        
        System.out.println("\n=== ÍNDICE DE MASA CORPORAL ===");
        System.out.println("Peso: " + peso + " kg");
        System.out.println("Altura: " + altura + " m");
        System.out.printf("IMC: %.2f%n", imc);
        
        // Clasificación
        String clasificacion;
        if (imc < 18.5) {
            clasificacion = "Bajo peso";
        } else if (imc < 25) {
            clasificacion = "Peso normal";
        } else if (imc < 30) {
            clasificacion = "Sobrepeso";
        } else {
            clasificacion = "Obesidad";
        }
        
        System.out.println("Clasificación: " + clasificacion);
        
        scanner.close();
    }
}

/*
 * SALIDA ESPERADA:
 * ===============
 * Digite su peso en kg: 70
 * Digite su altura en metros: 1.75
 * 
 * === ÍNDICE DE MASA CORPORAL ===
 * Peso: 70.0 kg
 * Altura: 1.75 m
 * IMC: 22.86
 * Clasificación: Peso normal
 */
